src/main/java/it/aboutbits/springboot/emailservice/lib/application/SendScheduledEmails.java

```diff
@@ -0,0 +1,50 @@
+package it.aboutbits.springboot.emailservice.lib.application;
+
+
+import it.aboutbits.springboot.emailservice.lib.EmailSchedulerCallback;
+import lombok.AllArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.scheduling.annotation.Scheduled;
+
+import java.util.List;
+
+@AllArgsConstructor
+@Log4j2
+public class SendScheduledEmails {
+    private static final String JOB_DESCRIPTION = "Sending open and failed email notifications.";
+
+    private final QueryEmail queryEmail;
+    private final ManageEmail manageEmail;
+    private final List<EmailSchedulerCallback> callbacks;
+
+    @Scheduled(initialDelayString = "${lib.emailservice.scheduling.interval:30000}", fixedDelayString = "${lib.emailservice.scheduling.interval:30000}")
+    void sendEmailNotifications() {
+        log.info("Start: " + JOB_DESCRIPTION);
```

Comment:
if you do not run the schedulers in a threadpool, they can never run in parallel.
also, if you do not use cron notation but like here `fixedDelay`(and not `fixedRate`) it will always wait for the job to end before starting it again.

Reply:
I'm referring to apps, that run in Kubernetes with two pods.